pkg/reconciler/route/domains/domains.go

```diff
@@ -63,6 +65,28 @@ func GetAllDomainsAndTags(ctx context.Context, r *v1.Route, names []string, visi
 	return domainTagMap, nil
 }
 
+// GetDomainsForVisibility return all domains for the specified visibility.
+func GetDomainsForVisibility(ctx context.Context, targetName string, r *v1.Route, visibility netv1alpha1.IngressVisibility) (sets.String, error) {
```

Comment:
this is moved and renamed from `ingress.go`

I also added some tests for this function.